channels/generic/http.py

```diff
@@ -78,11 +81,7 @@ def __init__(self, *args, **kwargs):
         if "body" in message:
             self.body.append(message["body"])
         if not message.get("more_body"):
-            try:
-                await self.handle(b"".join(self.body))
-            finally:
-                await self.disconnect()
-                raise StopConsumer()
+            await self.handle(b"".join(self.body))
```

Comment:
So simplifying just to `await self.handle(...)` the intention is that no matter what's going on with `message` (additional bytes in it or not), we want/need to let the runner (daphne etc) deal with/control the rest of the flow, naturally.

@carltongibson so this specific change is correct/ready?

Reply:
Hi @jheld. Yes, I think so. 

The reason I haven't pushed on here is that I think we need to make sure Daphne is able to correctly handle the case where we started the response and then error-ed after some time (thinking SSE or such). Currently it'll send an entire response, where we need it to just finish up the current one.  (Q: what would uvicorn do in that case?)

Tests for all that. 

Also I think we can document: "if your consumer is liable to raise an exception, you should probably add logic to handle that, otherwise you're falling back to default handling at the ASGI server level". (Or such)

(I think the other PRs here are essentially trying to build that logic into the generic consumers...) 

Make sense?